### PR TITLE
High level API fixes December 23

### LINKED ISSUE #579: [BUG] commissioningController autoconnect:true fails on startup
1. When starting up a commissioningController with empty storage and the property {autoConnect: true}, the program begins searching for previously connected nodes. If it cannot find one (due to starting with empty storage), the program throws an ImplementationError, whereas it should simply exit the async connect() function.
Screenshot from 2023-12-13 16-01-03

2. If the controller has information about previously connected nodes, with the same autoConnect: true field, it starts reconnecting to all the nodes upon restart. If it fails to connect to one (e.g., the devices went offline in the meantime), it will not proceed to the next node for reconnection.

### PR DESCRIPTION
**Pull request by Apollon77**
This PR addresses a few issueson the high level API reported by developers, see commits and changelog (Changelog also contains missed stuff from matter.js core)

fixes #579

### CODE REVIEW

File: packages/matter.js/src/MatterServer.ts [hunk 1]
```diff
@@ -224,7 +224,12 @@ export class MatterServer {
         }
         this.started = true;
         for (const node of this.nodes) {
-            await this.prepareNode(node);
+            try {
+                await this.prepareNode(node);
+            } catch (error) {
+                // TODO: Find a better way how to report back such issues and which nodes errored
```
Comment (lauckhart) [inline packages/matter.js/src/MatterServer.ts:230]: Is there any identifier we could include in error message from the node object that is in scope?

Could also maybe add a todo for retry unless that happens at a higher level.

**Review by lauckhart — APPROVED** [review]

File: packages/matter.js/src/MatterServer.ts (hunk 1 above)
Comment (Apollon77) [reply packages/matter.js/src/MatterServer.ts:230]: We might have the unique storage key or at least the index. Will add. Good idea

For connect -now that you state it - it should normally not error out but go into "wait" state. Need to check that

Comment (Apollon77) [reply packages/matter.js/src/MatterServer.ts:243]: @lauckhart See additional commit. I now also added uniqueness check for storagekey (which indeed was missing) ... and so we also have that to log. Additoonally: With current code exceptions should only occure on invalid config, so I would not do retry. If a node is commussioned then it should end in "waiting for device to come online for connection" and not throw an error

Comment (lauckhart) [reply packages/matter.js/src/MatterServer.ts:243]: Looks good

**Pull request merged**